pytest_bdd/scripts.py

```diff
@@ -26,7 +26,7 @@ def migrate_tests_in_file(file_path):
     try:
         with open(file_path, "r+") as fd:
             content = fd.read()
-            new_content = MIGRATE_REGEX.sub(r"\n@scenario(\2)\ndef \1():\n    pass\n", content)
+            new_content = MIGRATE_REGEX.sub(r"\n@scenario(\2)\ndef \1():\n    pass", content)
```

Comment:
I wasn't sure about this here, by adjusting the test to use `dedent` instead of `Source` it showed that `migrate` was adding an extra newline here (I wasn't sure if this was intentional or not but it was causing the file to end with `\n\n` when previously it did not)

Another alternative I could take here is: `contents = contents.rstrip('\n') + '\n'` to ensure that the file ends in a newline

let me know!

Reply:
actually, now that I look at this -- this'll cause

```python
foo = scenario(...)
bar = scenario(...)
```

to become

```python
@scenario(...)
def foo():
    pass
@scenario(...)
def bar():
    pass
```

instead of

```python
@scenario(...)
def foo():
    pass

@scenario(...)
def bar():
    pass
```

I think I'll leave this regex and adjust the newlines at the end of the file